torch/nn/modules/loss.py

```diff
@@ -1246,6 +1251,121 @@ def forward(self, anchor: Tensor, positive: Tensor, negative: Tensor) -> Tensor:
                                      eps=self.eps, swap=self.swap, reduction=self.reduction)
 
 
+class TripletMarginWithDistanceLoss(_Loss):
+    r"""Creates a criterion that measures the triplet loss given input
+    tensors :math:`a`, :math:`p`, and :math:`n` (representing anchor,
+    positive, and negative examples, respectively); and a nonnegative,
+    real-valued function ("distance function") used to compute the relationship
+    between the anchor and positive examples ("positive distance") and the
```

Comment:
"the positive example? -- only a single positive (and negative, later in this sentence) example